### PR TITLE
Remove javax.* dependencies

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by olamy** [comment]
fix for issue #5754

File: pom.xml [hunk 1]
```diff
@@ -151,11 +151,36 @@
...
+            <id>ban-javax-servlet-api</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <phase>validate</phase>
+            <configuration>
+              <rules>
+                <bannedDependencies>
+                  <excludes>
+                    <exclude>javax.servlet:*</exclude>
```
Comment (janbartel) [inline pom.xml:166]: I don't understand: the name of this enforcement is `ban-javax-servlet-api`, but pretty much every permutation of `javax.servlet.*` is excluded?

**Review by janbartel — CHANGES_REQUESTED** [review]
There's plenty more references to `javax.servlet` in dependency excludes in other poms.

File: pom.xml (hunk 1 above)
Comment (joakime) [reply pom.xml:166]: That is a maven coordinate ban, not a java package ban.
Notice the `:`?  
The format of that line is `groupId[:artifactId][:version][:type][:scope][:classifier]`
So this usage is just banning the maven groupId of `javax.servlet` with any artifact/version/type etc ...

Comment (janbartel) [reply pom.xml:166]: Ah, I see the confusion: I thought the `<exclude>` was enumerating groupId.artifactIds to _exclude_ from the ban, whereas you're saying they are enumerating things to _include_!

Comment (joakime) [reply pom.xml:166]: Per the rule: the `<exclude>` list is what is banned, the `<include>` list are exceptions to what is banned.

**Review by joakime — APPROVED** [review]
I like this change.

**Pull request merged**